slatedb/src/db/builder.rs

```diff
@@ -174,9 +178,23 @@ impl<P: Into<Path>> DbBuilder<P> {
         self
     }
 
-    /// Sets the clock to use for the database.
-    pub fn with_clock(mut self, clock: Arc<dyn Clock + Send + Sync>) -> Self {
-        self.clock = Some(clock);
+    /// Sets the clock to use for user facing database operations.
+    /// Operations like writing or removing data from the database will use this clock
+    /// to store the timestamp of the operation.
+    ///
+    /// If not set, SlateDB uses the system clock.
+    pub fn with_user_clock(mut self, clock: Arc<dyn UserClock>) -> Self {
+        self.user_clock = Some(clock);
+        self
+    }
+
+    /// Sets the clock to use for system operations.
+    /// System operations are operations that are not user facing, like compaction, garbage collection,
+    /// and WAL flushing. These operations are not affected by the user clock.
```

Comment:
I think it's worth calling out that this should always use a wall clock and is only expected to be set in situations like DST. You can really shoot yourself in the foot by setting this to a clock that may not progress in certain situations.

I'm even tempted to suggest making it `pub(crate)` just for our DST but I guess it's useful if someone wants to test a system that uses SlateDB with DST themselves...

cc @criccomini